pennylane/operation.py

```diff
@@ -480,6 +480,67 @@ def matrix(self, wire_order=None):
 
         return expand_matrix(canonical_matrix, wires=self.wires, wire_order=wire_order)
 
+    @staticmethod
+    def compute_sparse_matrix(*params, **hyperparams):  # pylint:disable=unused-argument
+        """Canonical matrix of this operator in the computational basis, using a sparse
+        matrix type.
+
+        The canonical matrix is the textbook matrix representation that does not consider wires.
+        Implicitly, this assumes that the wires of the operator correspond to the global wire order.
+
+        .. note::
+            This method gets overwritten by subclasses to define the sparse matrix representation
+            of a particular operator.
+
+        Args:
+            params (list): trainable parameters of this operator, as stored in ``op.parameters``
+            hyperparams (dict): non-trainable hyperparameters of this operator, as stored in ``op.hyperparameters``
+
+        Returns:
+            scipy.sparse.coo.coo_matrix: matrix representation
```

Comment:
Curious, how come we specify COO matrix here?

Should we leave the return type up to the method, like we do with `matrix`? E.g., `sparse_like`?

Having said that, there is logic in default.qubit currently which assumes COO matrices, so that is already baked into the codebase...

Reply:
Let's be specific for now, and hope to relax later!